packages/web-app/app/_server/active-campaign.ts

```diff
@@ -0,0 +1,109 @@
+'use server';
+import { z } from 'zod';
+
+process.env.NODE_TLS_REJECT_UNAUTHORIZED = '0';
+
+const options = {
+  method: 'POST',
+  headers: {
+    accept: 'application/json',
+    'content-type': 'application/json',
+    'Api-Token': process.env.NEXT_ACTIVE_CAMPAIGN_API_KEY,
+  },
+};
+
+const createContact = async (email: string) => {
+  const parsedEmail = z.string().safeParse(email);
+
+  if (!parsedEmail.success) {
+    return {
+      error: 'Invalid email',
+    };
+  }
+
+  const validEmail = parsedEmail.data;
+
+  try {
+    const response = await fetch(
+      'https://fractal12658.api-us1.com/api/3/contacts',
+      {
+        ...options,
+        body: JSON.stringify({ contact: { email: validEmail } }),
+      },
+    );
+
+    const result = await response.json();
+    const contactId = result?.contact?.id;
+
+    const contactIdParsed = z.string().safeParse(contactId);
+
+    if (!contactId || !contactIdParsed.success) {
+      return { error: 'Failed to create contact' };
+    }
+
+    return contactIdParsed.data;
+  } catch (error) {
+    console.error(error);
+
+    if (error instanceof Error) {
+      console.error(error.message);
+      return { error: error.message };
+    }
+
+    return { error: 'Failed to add to mailing list' };
+  }
+};
+
+const updateListState = async (contactId: string) => {
+  const parsedContactId = z.string().safeParse(contactId);
+
+  if (!parsedContactId.success) {
+    return {
+      error: 'Invalid contact id',
+    };
+  }
+
+  const validContact = parsedContactId.data;
+
+  try {
+    const response = await fetch(
+      'https://fractal12658.api-us1.com/api/3/contactLists',
```

Comment:
Is it ok for this to be hardcoded or should we fetch from an env var?

Reply:
I think it is ok as the secret is already in an env var, but I can move the id too